README.rst

```diff
@@ -454,16 +454,184 @@ template ``context_processors`` in ``settings.py`` as shown below.
+
+.. code-block:: python
+
+    # use to create a link of list page
+    my_model_list_config = {
+        'model': 'my_project.MyModel',
+        'name': 'changelist',
+        'label': 'MyModel Change List',
+        'icon': 'my-model-list-class',
+    }
+
+    #use to create a link of add page
+    my_model_add_config = {
+        'model': 'my_project.MyModel',
+        'name': 'add',
+        'label': 'MyModel Add Item',
+        'icon': 'my-model-add-class',
+    }
+    register_menu_group(position=1, config=my_model_list_config)
+    register_menu_group(position=2, config=my_model_add_config)
+
+Following is the description of the configuration:
+
++------------------+--------------------------------------------------------------+
+| **Parameter**    | **Description**                                              |
++------------------+--------------------------------------------------------------+
+| ``model``        | (``str``) Model of a app.  (``required``)                    |
++------------------+--------------------------------------------------------------+
+| ``name``         | (``str``) url name. eg. changelist or add.  (``required``)   +
++------------------+--------------------------------------------------------------+
+| ``label``        | (``str``) Display name for the link.  (``optional``)         |
++------------------+--------------------------------------------------------------+
+| ``icon``         | (``str``) CSS class name of the link icon.  (``optional``)   |
++------------------+--------------------------------------------------------------+
+
+``Create a menu group``
+^^^^^^^^^^^^^^^^^^^^^^^
+
+If you want to create a nested menu group then you can use following syntax.
+It creates a dropdown in the menu.
+
+**Syntax:**
+
+.. code-block:: python
+    
+    my_group_items = {
+        1: {'label': 'Link Label', 'url': 'link_url', 'icon': 'my-icon'},
+        2: {
+            'model': 'my_project.MyModel',
+            'name': 'changelist',
+            'label': 'MyModel Change List',
+            'icon': 'my-model-list-class',
+            }
+    }
+    register_menu_group(
+        position=1,
+        config={'label': 'My Label', 'items': my_group_items, 'icon': 'my-icon-class'},
+    )
+
+Following is the description of the configuration:
+
++------------------+--------------------------------------------------------------+
+| **Parameter**    | **Description**                                              |
++------------------+--------------------------------------------------------------+
+| ``label``        | (``str``) Display name for the link.  (``required``)         |
++------------------+--------------------------------------------------------------+
+| ``items``        | (``dict``) Items to be displayed in the dropdown.            |
+|                  | It can be a dict of ``general link`` or ``model link``       |
+|           .      | with key as their position.                                  |
```

Comment:
something is wrong here, please look at the generated output: https://github.com/niteshsinha17/openwisp-utils/tree/gsoc/new-navigation-menu-and-ux-improvements#create-a-menu-group
It does not look right. Please after writing docs always look at the generated output and ensure there are no quirks.

Reply:
Fixed